### PR TITLE
Update Service Mesh docs

### LINKED ISSUE #27167: Document cilium-operator role in Service Mesh
Gateway API doesn't seem to be documented here, but the operator plays a role in implementing Gateway API:

https://docs.cilium.io/en/latest/internals/cilium_operator/#cilium-operator

```[tasklist]
### Tasks
- [ ] Document the operator's role in Gateway API
- [ ] Document Ingress role
- [ ] Document Mutual Authentication role
```

### PR DESCRIPTION
**Pull request by youngnick**
Updates Service mesh docs to add some more explanation and cover some features more correctly. Each change is in a separate commit.

I was fixing #27167 when I noticed some other things that were missing, so I've hit all the ones I can find while I'm at it.

Fixes: #27167

### CODE REVIEW

File: Documentation/internals/cilium_operator.rst [hunk 1]
```diff
@@ -220,3 +220,22 @@ constructs to derivative CNP/CCNP objects without these fields.
...
+a Cilium-managed namespace that the Cilium Agent is then granted access to. This
+reduces the permissions required of the Cilium Agent.
+
+Mutual Authentication Support
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When Cilium's Mutual Authentication Support is enabled, the Cilium Operator is
+responsible for ensuring that each Cilium Identity has an associated identity
+in the certificate management system (Currently only SPIRE is supported). It will
+create and delete SPIRE identity registrations as required.
```
Comment (meyskens) [inline Documentation/internals/cilium_operator.rst]: smal nit could we add that he operator itself has no access to the key material in the identities?

File: Documentation/network/servicemesh/l7-traffic-management.rst [hunk 2]
```diff
@@ -16,11 +16,23 @@ and CiliumClusterwideEnvoyConfig).
...
-  or strict. Please refer to :ref:`kube-proxy replacement <kubeproxy-free>`
+* Cilium must be configured with ``kubeProxyReplacement`` as ``true``.
+  Please refer to :ref:`kube-proxy replacement <kubeproxy-free>`
   for more details.
 * The minimum supported Kubernetes version for Ingress is 1.19.
 
+Caveats
+#######
+
+* ``CiliumEnvoyConfig`` resources have only minimal validation performed, and
```
Comment (meyskens) [inline Documentation/network/servicemesh/l7-traffic-management.rst:27]: Could we add some kind of warning that it is also super easy to just pass incorrect config and have it ingested without proper error response to the user? (but in a more friendly way than i said it)

**Review by meyskens — APPROVED** [review]
Small nit and a suggestion but overall an LGTM for me

**Review by squeed — APPROVED** [review]

File: Documentation/network/servicemesh/l7-traffic-management.rst (hunk 2 above)
Comment (youngnick) [reply Documentation/network/servicemesh/l7-traffic-management.rst:27]: Well, I did my best on this one. 😄

File: Documentation/internals/cilium_operator.rst [hunk 3]
```diff
@@ -220,3 +220,24 @@ constructs to derivative CNP/CCNP objects without these fields.
...
+
+When Ingress or Gateway API support is enabled, the Cilium Operator performs the
+task of parsing Ingress or Gateway API objects and converting them into
+``CiliumEnvoyConfig`` objects used for configuring the per-node Envoy proxy.
+
+Additionally, Secrets used by Ingress or Gateway API objects will be synced to
+a Cilium-managed namespace that the Cilium Agent is then granted access to. This
+reduces the permissions required of the Cilium Agent.
+
+Mutual Authentication Support
```
Comment (learnitall) [inline Documentation/internals/cilium_operator.rst:235]: Do you think you could reword this section so it doesn't mention SPIRE? And have it be more generic towards whichever certificate management system is used? That way if we update the supported list of systems we don't have to come back to this page and make an update.

File: Documentation/network/servicemesh/ingress.rst [hunk 4]
```diff
@@ -106,7 +106,21 @@ Supported Ingress Annotations
...
+By default, annotations with values beginning with:
+
+* ``service.beta.kubernetes.io``
+* ``service.kubernetes.io``
+* ``cloud.google.com``
+
+will be copied from an Ingress object to the generated LoadBalancer service objects.
+
+This setting is controlled by the ``ingress-lb-annotation-prefixes`` config item
+for Cilium's operator, and can be configured in Cilium's Helm ``values.yaml``
```
Comment (learnitall) [inline Documentation/network/servicemesh/ingress.rst]: Small nit on wording for clarity.

```suggestion
This setting is controlled by the Cilium Operator's ``ingress-lb-annotation-prefixes``
config flag, and can be configured in Cilium's Helm ``values.yaml``
```

File: Documentation/network/servicemesh/l7-traffic-management.rst [hunk 5]
```diff
@@ -16,11 +16,30 @@ and CiliumClusterwideEnvoyConfig).
...
+  do not have a defined conflict resolution behavior. This means that if you
+  create multiple CECs that modify the same parts of Envoy's config, the results
+  may be unpredictable.
+* In addition to this minimal validation, ``CiliumEnvoyConfig`` has minimal
+  feedback to the user about the correctness of the configuration. So in the
```
Comment (learnitall) [inline Documentation/network/servicemesh/l7-traffic-management.rst:32]: Small grammar nit

```suggestion
  feedback to the user about the correctness of the configuration. In the
```

File: Documentation/network/servicemesh/l7-traffic-management.rst [hunk 6]
```diff
@@ -16,11 +16,30 @@ and CiliumClusterwideEnvoyConfig).
...
+  event a CEC does produce an undesirable outcome, troubleshooting will require
+  inspecting the Envoy config and logs, rather than being able to look at the
+  ``CiliumEnvoyConfig`` in question.  
+* ``CiliumEnvoyConfig`` is used by Cilium's Ingress and Gateway API support to
+  direct traffic through the per-node Envoy proxies. If you create CECs that
+  conflict with or modify the autogenerated config, results may be unpredictable.
+  Be very careful using CECs for these use cases. (The above risks are managed
+  by ensuring that all config generated by Cilium is semantically valid, as far
+  as possible).
```
Comment (learnitall) [inline Documentation/network/servicemesh/l7-traffic-management.rst]: Would it be possible to remove the parentheticals here?

**Review by learnitall — CHANGES_REQUESTED** [review]
Well done! Really well written! I just have a couple of small nits and suggestions for you.

File: Documentation/internals/cilium_operator.rst (hunk 3 above)
Comment (youngnick) [reply Documentation/internals/cilium_operator.rst:235]: Good idea, thanks!

File: Documentation/network/servicemesh/l7-traffic-management.rst (hunk 6 above)
Comment (youngnick) [reply Documentation/network/servicemesh/l7-traffic-management.rst]: Good call, the parentheses were adding nothing, removed.

**Review by learnitall — APPROVED** [review]
Awesome looks good!

**Comment by learnitall** [comment]
Hey @youngnick, two questions for you:

* Do you mind squashing commits before merging?
* Would you mind handling the backports for 1.13 (and 1.12 if needed)? I'm not super familiar with all of these features and their timelines, but I believe that only some of the changes in this PR need to be backported to pre-1.14, correct?

**Comment by youngnick** [comment]
Yes, no problem about squashing (although I'll keep another branch with the individual commits to make those backports easier).

And yes, I can handle backporting.

**Pull request merged**